metricsaggregator/src/main/java/io/dockstore/metricsaggregator/client/cli/MetricsAggregatorClient.java

```diff
@@ -184,6 +186,7 @@ private void submitValidationData(MetricsAggregatorConfig config, ValidatorToolE
                     .validatorToolVersion(validatorVersion)
                     .isValid(isValid);
             validationExecution.setDateExecuted(dateExecuted);
+            validationExecution.setExecutionId(UUID.randomUUID().toString()); // No execution ID was provided by DNAstack, generate a random one
```

Comment:
Hmm, this command only uploads to one environment, so it's currently not possible to keep the execution ID consisten between QA, staging, and prod for this command

Reply:
I meant, would it make sense to provide it as a parameter like `java -jar target/metricsaggregator-*-SNAPSHOT.jar submit-validation-data --config my-custom-config --data <path-to-my-data-file> --validator MINIWDL --validatorVersion 1.0 --platform DNA_STACK --execution-id first-run-after-ai-omics-release` for example

Reply:
Ah I see, hmm yes I think that would work assuming that the file DNAstack provided us contains no duplicate combo of TRS ID and version, otherwise the duplicates will fail because the execution ID already exists for the TRS ID and version (and platform). 

I think it's a safe assumption to make given that they provided us with a list of TRS IDs and versions that validated correctly with miniwdl (there's no point in duplicate TRS ID and version entries because it conveys the same info). I'll make the change so it's consistent between environments

Reply:
I may be getting turned around, could you refresh my memory as to how the execution id enters the system and is there a difference between validations and normal executions in light of 

```
In order to efficiently find the execution to update, the S3 key must contain the execution ID. The file name in S3 was previously the current time in ms, but it's now the user-provided execution ID. This means that each file in the metrics S3 bucket contains only 1 execution, whereas it previously could contain multiple. 
```

Reply:
> could you refresh my memory as to how the execution id enters the system

In https://github.com/dockstore/dockstore/pull/5778, the execution ID enters the system through the [POST /api/ga4gh/v2/extended/{id}/versions/{version_id}/executions](https://qa.dockstore.org/api/static/swagger-ui/index.html#/extendedGA4GH/executionMetricsPost) endpoint. In that endpoint's request body, users specify the executions that they want to submit, whether it's a list of workflow executions, a list of validation executions, and/or a list of taskExecutions sets. Each execution in each list of executions requires a user-provided ID that will become the file name in S3.

> is there a difference between validations and normal executions in light of
     _In order to efficiently find the execution to update, the S3 key must contain the execution ID. The file name in S3 was previously the current time in ms, but it's now the user-provided execution ID. This means that each file in the metrics S3 bucket contains only 1 execution, whereas it previously could contain multiple._

There is no difference between validations and normal executions - validations are a type of execution and are included in the italicized comment, i.e. they need a user-provided execution ID and they can be updated. There's 3 types of executions: workflow executions, validation executions, and task executions